src/models/sweeper.rs

```diff
@@ -88,39 +88,48 @@ impl Sweeper {
         let cutoff_time = Utc::now() - self.max_message_age;
 
         let success_count = Arc::new(AtomicU32::new(0));
-
+        let delete_message_ids: Arc<Mutex<Vec<MessageId>>> = Arc::new(Mutex::new(vec![]));
```

Comment:
You should be able to import `futures::stream::StreamExt` and then use [collect](https://rust-lang-nursery.github.io/futures-api-docs/0.3.0-alpha.5/futures/stream/trait.StreamExt.html#method.collect) there.

Will change the code below to something like
```rust
let message_ids: Vec<MessageId> = self
    .message_stream()
    .try_take_while(...)
    .map(|message| message.id)
    .collect()
    .await;

for chunk in message_ids.chunk(100) { // 100 at a time
    self.channel_id.delete_messages(chunk).await
}
```

There _might_ be some horrible trait wrangling that needs to happen for the `collect` to work but I don't think it should be too terrible.